util/include/cppmicroservices/util/DataContainer.h

```diff
@@ -0,0 +1,68 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/CppMicroServices/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef CPPMICROSERVICES_DATACONTAINER_H
+#define CPPMICROSERVICES_DATACONTAINER_H
+
+#include "cppmicroservices/GlobalConfig.h"
+
+#include <cstring>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace cppmicroservices {
+
+// A generic abstract base class to store
+// and retrieve raw data bytes
+class DataContainer
```

Comment:
I think all classes derived from `DataContainer` are non-copyable. You can instead `delete` the copy ctor and assignment here to enforce that `DataContainer` cannot be copied. 

And remove the copy ctor & assignment deletions from the derived classes.

Reply:
I've modified the code to make these move only objects.